crates/cheatcodes/src/inspector.rs

```diff
@@ -1250,6 +1254,37 @@ impl Cheatcodes {
             return None;
         }
 
+        let mut create2_factory_deps = Vec::new();
+
+        if call.target_address == DEFAULT_CREATE2_DEPLOYER && self.use_zk_vm {
+            call.target_address = DEFAULT_CREATE2_DEPLOYER_ZKSYNC;
+            call.bytecode_address = DEFAULT_CREATE2_DEPLOYER_ZKSYNC;
+
+            let (salt, init_code) = call.input.split_at(32);
+            let contract = self
+                .dual_compiled_contracts
+                .find_by_evm_bytecode(init_code)
+                .unwrap_or_else(|| panic!("failed finding contract for {init_code:?}"));
+
+            create2_factory_deps.push(contract.zk_deployed_bytecode.clone());
+
+            let factory_deps = self.dual_compiled_contracts.fetch_all_factory_deps(contract);
+
+            // This is a hack to pass it to the call
+            self.persisted_factory_deps
```

Comment:
since you have the variable outside just use that instead

Reply:
actually, right now the variable will only have 1 fdep, at least when it's used for the broadcastable tx, which is a bug

Reply:
Why are we extending `persisted_factory_deps` directly here? This field is currently populated by the zkEVM automatically and shouldn't exactly require manual intervention.

Reply:
Can we not have this entire logic in `try_call_in_zk` instead? That would contain all of this extra logic without the need to pass factory deps. We do similar for `try_create_in_zk`.